mountpoint-s3/tests/fuse_tests/fork_test.rs

```diff
@@ -484,6 +486,221 @@ fn mount_with_sse(
+    let (bucket, prefix) = get_test_bucket_and_prefix("mount_with_assumed_role_no_region");
+    let subsession_role = get_subsession_iam_role();
+    let region = get_test_region();
+    let mount_point = assert_fs::TempDir::new()?;
+    let profile_name = "fork_test";
+    let source_profile = "default";
+
+    let config_file = create_cli_config_file(profile_name, source_profile, &subsession_role, &region)?;
+
+    // Credentials are provided as environment variables when running on GitHub Actions,
+    // so we will have to build a credentials file from those variables if they exist, otherwise
+    // we are going to use default profile from the test instance.
+    let credentials_file = create_shared_credentials_file_if_possible(source_profile)?;
+
+    let mut cmd = Command::cargo_bin("mount-s3")?;
+    if let Some(file) = &credentials_file {
+        cmd.env("AWS_SHARED_CREDENTIALS_FILE", file.path());
+    }
+    let child = cmd
+        .arg(&bucket)
+        .arg(mount_point.path())
+        .arg(format!("--prefix={prefix}"))
+        .arg("--auto-unmount")
+        .env("AWS_CONFIG_FILE", config_file.path())
+        .env("AWS_PROFILE", profile_name)
+        .env_remove("AWS_ACCESS_KEY_ID")
+        .env_remove("AWS_SECRET_ACCESS_KEY")
+        .env_remove("AWS_SESSION_TOKEN")
+        .env("AWS_EC2_METADATA_DISABLED", "true")
+        .env_remove("AWS_REGION")
+        .env_remove("AWS_DEFAULT_REGION")
+        .spawn()
+        .expect("unable to spawn child");
+
+    let exit_status = wait_for_exit(child);
+
+    // verify mount status and mount entry
+    assert!(exit_status.success());
+    assert!(mount_exists("mountpoint-s3", mount_point.path().to_str().unwrap()));
+
+    test_read_files(&bucket, &prefix, &region, &mount_point.to_path_buf());
+
+    unmount(mount_point.path());
+
+    Ok(())
+}
+
+#[test]
+fn run_fail_when_assume_role_with_invalid_arn() -> Result<(), Box<dyn std::error::Error>> {
+    let (bucket, prefix) = get_test_bucket_and_prefix("run_fail_when_assume_role_with_invalid_arn");
+    let invalid_role = "arn:aws:iam::123456789123:role/invalid-role";
+    let region = get_test_region();
+    let mount_point = assert_fs::TempDir::new()?;
+    let profile_name = "fork_test";
+    let source_profile = "default";
+
+    let config_file = create_cli_config_file(profile_name, source_profile, invalid_role, &region)?;
+
+    // Credentials are provided as environment variables when running on GitHub Actions,
+    // so we will have to build a credentials file from those variables if they exist, otherwise
+    // we are going to use default profile from the test instance.
+    let credentials_file = create_shared_credentials_file_if_possible(source_profile)?;
+
+    // First, make sure we can mount with the default profile
+    let mut cmd = Command::cargo_bin("mount-s3")?;
+    if let Some(file) = &credentials_file {
+        cmd.env("AWS_SHARED_CREDENTIALS_FILE", file.path());
+    }
```

Comment:
Related to the other comment:

I'm hoping each test just becomes:
- Populate source profile
- Populate profile under test (which draws from source)
- Configure only `AWS_CONFIG_FILE`